### PR TITLE
Add Validation and TextField components

### LINKED ISSUE #24: Validation component
Related to ANXR-193.

### PR DESCRIPTION
**Pull request by jondlm**
Fixes #24 and #23  by adding two new components `Validation` and `TextField`.

### CODE REVIEW

File: src/components/TextField/TextField.less [hunk 1]
```diff
@@ -0,0 +1,51 @@
+@TextField-size-padding: 8px;
+@TextField-size-height: 28px;
+@TextField-color-font: #333;
+
+.lucid-TextField {
+	&-native {
+		.box-sizing();
+		.inset-box-shadow();
+
+		background-color: #fff;
```
Comment (alisonailea) [inline src/components/TextField/TextField.less]: this should probably be a var

File: src/components/TextField/TextField.less [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+	&-is-multi-line &-native {
+		vertical-align: top;
+		resize: both;
+		padding-top: @TextField-size-padding - 2px; // Handle padding appropriately for `textarea`s
+	}
+
+	&-is-disabled &-native {
+		cursor: not-allowed;
+		background-color: fade(#000, 5%);
```
Comment (alisonailea) [inline src/components/TextField/TextField.less]: these should be variables

File: src/styles/mixins.less [hunk 3]
```diff
@@ -1,46 +1,46 @@
...
-    z-index: -100;
-    transition: opacity @transTime, box-shadow .05s;
+	position: relative;
+	background-size: 100%;
+	background-image: linear-gradient(@start, @end);
+	// todo: create zindex pattern
+	z-index: 100;
+	&:before {
+		background-image: linear-gradient(@end, @start);
+		background-color: #fff;
```
Comment (alisonailea) [inline src/styles/mixins.less]: variable

File: src/components/TextField/TextField.jsx [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+const boundClassNames = bindClassNames('TextField');
+
+const {
+	bool,
+	string,
+	func,
+	number,
+	object,
+	any,
```
Comment (nathaniellee) [inline src/components/TextField/TextField.jsx]: Extraneous comma? :)

File: src/components/TextField/TextField.jsx [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+	propTypes: {
+		/**
+		 * Styles that are passed through to the root `div` container.
+		 */
+		style: object,
+
+		/**
+		 * Styles that are passed through to the underlying `input`.
+		 */
+		inputStyle: object,
```
Comment (nathaniellee) [inline src/components/TextField/TextField.jsx]: Is this something we should be doing on our native control replacements?

**Comment by nathaniellee** [comment]
Looks cool to me...

File: src/components/TextField/TextField.jsx (hunk 5 above)
Comment (jondlm) [inline src/components/TextField/TextField.jsx]: Yeah, I'm not really sure. I'd love for @ogupte to weigh in as well. It seemed like a good idea?

File: src/components/TextField/TextField.jsx (hunk 4 above)
Comment (jondlm) [inline src/components/TextField/TextField.jsx]: I've been adding commas lately to the last item so I can sort later. I'm okay if people don't like this.

Comment (ogupte) [inline src/components/TextField/TextField.jsx]: i don't mind trailing commas, but i know some really hate it, and i just don't like inconsistency

File: src/components/TextField/TextField.jsx (hunk 5 above)
Comment (ogupte) [inline src/components/TextField/TextField.jsx]: hard to say...you could easily target the native `input` with css, but if you are using all inline styles, this prop helps, but it doesn't go far enough in all cases.

File: src/components/Validation/Validation.jsx [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+ *
+ * Validation is a wrapper component that's meant to be used by other
+ * components. Wrap your form components in it and style them accordingly if
+ * there's an error.
+ */
+const Validation = React.createClass(createLucidComponentDefinition({
+	displayName: 'Validation',
+
+	childProps: {
+		Error: {} // intentionally left blank since we only care about `children`
```
Comment (ogupte) [inline src/components/Validation/Validation.jsx]: we can probably make this use case better in the component definition, `null` also works here, but might look weird

File: src/index.js [hunk 7]
```diff
@@ -22,8 +24,8 @@ export default {
 	Button,
 	ButtonGroup,
 	CaretIcon,
-	Checkbox,
 	CheckIcon,
+	Checkbox,
```
Comment (ogupte) [inline src/index.js]: i added this in a different PR

**Comment by ogupte** [comment]
approved.

File: src/components/Validation/Validation.jsx (hunk 6 above)
Comment (jondlm) [inline src/components/Validation/Validation.jsx]: Meh, I'm okay with it as is. `null` works too, that's cool.

File: src/index.js (hunk 7 above)
Comment (jondlm) [inline src/index.js]: Whoops, I don't know why I threw this in :/

**Pull request merged**